### DISCUSSION TITLE
How to virtual register with default value?

### DISCUSSION THREAD

**Discussion by Noitidart** [react-hook-form/react-hook-form#6735] [Q&A]
Is it possible to virtually register a field with default value?

I do this:

```
useForm({
  groupsById: {
     43: {
         lineItemsById: {
              100: { name: 'foo' }
         }
     }
});
```

But when I want to delete line-item with id 100 I do: `form.unregister('groupsById.43.lineItemsById.100')` but this completely empties my values to: `{}`.

Is it possible to set default value for `groupById` so that when I unregister the line-item like above, it will leave values as `{ groupsById: { 43: {} }`?

I thought to do a:

```
useEffect(() => {
   form.register('groupsById')
})
```

But I'm not sure how to pass defaultValue to `form.register`.

Here is codesandbox demo, you will notice you can click "Add Group" which adds, but then clicking "Delete Group" makes values to go to `{}`:

https://codesandbox.io/s/delete-empties-values-react-hook-form-r9s49?file=/src/App.js:84-102

**Comment by bluebill1049** [comment]
```
useEffect(() => {
   register('groupsById', { value: '1235' })
}, [register])
```

https://react-hook-form.com/api/useform/register
<img width="952" alt="Screen Shot 2021-10-08 at 10 18 11 am" src="https://user-images.githubusercontent.com/10513364/136474987-a4d8a030-bef8-4990-90f0-58f03e117c7e.png">

**Reply by Noitidart** [reply]
Thanks but this doesn't fix the problem of wiping out the entire object when it has no children.

Please try my example:

https://codesandbox.io/s/delete-empties-values-react-hook-form-r9s49?file=/src/App.js

After clicking add we see this:

<img width="406" alt="image" src="https://user-images.githubusercontent.com/6372489/136479516-d0927ada-8765-47b5-a73c-8a6afc791317.png">

After clicking remove we see this:

<img width="333" alt="image" src="https://user-images.githubusercontent.com/6372489/136479527-f21bf655-8fef-4061-9685-ca03ebe8229a.png">

But I expected to see:

```
{
  "groups": {}
}
```

**Reply by Noitidart** [reply]
Oh someone on stackoverflow replied, is what they are saying true? https://stackoverflow.com/a/69487593/1828637

> RHF will remove the whole object if you unregister the last property inside it:
